lib/logstash/outputs/syslog.rb

```diff
@@ -84,71 +87,65 @@ class LogStash::Outputs::Syslog < LogStash::Outputs::Base
 
   # message text to log
   config :message, :validate => :string, :default => "%{message}"
- 
+
   # message id for syslog message
   config :msgid, :validate => :string, :default => "-"
 
   # syslog message format: you can choose between rfc3164 or rfc5424
   config :rfc, :validate => ["rfc3164", "rfc5424"], :default => "rfc3164"
 
-  
-  public
   def register
-      @client_socket = nil
-  end
-
-  private
-  def udp?
-    @protocol == "udp"
-  end
+    @client_socket = nil
 
-  private
-  def rfc3164?
-    @rfc == "rfc3164"
-  end 
+    facility_code = FACILITY_LABELS.index(@facility)
+    severity_code = SEVERITY_LABELS.index(@severity)
+    @priority = (facility_code * 8) + severity_code
 
-  private
-  def connect
-    if udp?
-        @client_socket = UDPSocket.new
-        @client_socket.connect(@host, @port)
-    else
-        @client_socket = TCPSocket.new(@host, @port)
-    end
+    # use instance variable to avoid string comparison for each event
+    @is_rfc3164 = (@rfc == "rfc3164")
   end
 
-  public
   def receive(event)
-    
-
     appname = event.sprintf(@appname)
     procid = event.sprintf(@procid)
     sourcehost = event.sprintf(@sourcehost)
 
-    facility_code = FACILITY_LABELS.index(@facility)
-
-    severity_code = SEVERITY_LABELS.index(@severity)
-
-    priority = (facility_code * 8) + severity_code
-
-    if rfc3164?
+    if @is_rfc3164
       timestamp = event.sprintf("%{+MMM dd HH:mm:ss}")
-      syslog_msg = "<"+priority.to_s()+">"+timestamp+" "+sourcehost+" "+appname+"["+procid+"]: "+event.sprintf(@message)
+      syslog_msg = "<" + @priority.to_s() + ">" + timestamp + " " + sourcehost + " " + appname + "[" + procid + "]: " + event.sprintf(@message)
```

Comment:
It's actually not very clear when interpolation wins vs concatenation. I have made some tests at some point and I think @jordansissel did too. IIRC its somehow related to the number of variables to bind.